pkg/collectors/managedclusterinfo.go

```diff
@@ -0,0 +1,132 @@
+package collectors
+
+import (
+	"context"
+
+	ocinfrav1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kube-state-metrics/pkg/metric"
+
+	mciv1beta1 "github.com/open-cluster-management/multicloud-operators-foundation/pkg/apis/internal.open-cluster-management.io/v1beta1"
+	"k8s.io/klog/v2"
+)
+
+var (
+	descClusterInfoName          = "ocm_managedcluster_info"
+	descClusterInfoHelp          = "Managed cluster information"
+	descClusterInfoDefaultLabels = []string{"hub_cluster_id", "cluster_id", "cluster", "vendor", "cloud", "version"}
+
+	cdGVR = schema.GroupVersionResource{
+		Group:    "hive.openshift.io",
+		Version:  "v1",
+		Resource: "clusterdeployments",
+	}
+
+	cvGVR = schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: "clusterversions",
+	}
+
+	mciGVR = schema.GroupVersionResource{
+		Group:    "internal.open-cluster-management.io",
+		Version:  "v1beta1",
+		Resource: "managedclusterinfos",
+	}
+)
+
+func getHubClusterId(c dynamic.Interface) string {
+
+	cvObj, errCv := c.Resource(cvGVR).Get(context.TODO(), "version", metav1.GetOptions{})
+	if errCv != nil {
+		klog.Fatalf("Error getting cluster version %v \n", errCv)
+		panic(errCv.Error())
```

Comment:
I think you would not need panic if fatal in advance.

Reply:
???
this is evaluated at startup... so if panic the pod will restart until fixed.

Reply:
I think klog.Fatal already makes it exit? https://godoc.org/k8s.io/klog#Fatal

Reply:
oh... of course